likeminds-feed-reactnative-integration/components/LMMedia/LMVideo/index.tsx

```diff
@@ -109,17 +112,41 @@ const LMVideo = React.memo(
       setDimensions({ width, height });
     };
 
+    const handleError = (error) => {
+      console.warn('Video playback error:', error);
+      setError(true);
+      // Retry after a delay
+      if (retryCount < 3) {
+        setTimeout(() => {
+          setRetryKey(prev => prev + 1);
+          setRetryCount(prev => prev + 1);
+        }, 2000);
+      }
+    };
+
     useEffect(() => {
-      const ScreenWidth = Dimensions.get("window").width;
-      const desiredAspectRatio = width > height ? 1.91 : 0.8;
-      const heightCalculated = ScreenWidth * (1 / desiredAspectRatio);
-      setHeightCalculated(heightCalculated);
-      setDesiredAspectRatio(desiredAspectRatio);
+      if ( (!height || !width) && (dimensions?.height > 0 && dimensions?.width > 0)) {
+
+        const ScreenWidth = Dimensions.get("window").width;
+        const desiredAspectRatio = dimensions?.width > dimensions?.height ? 1.91 : 0.8;
+        const heightCalculated = ScreenWidth * (1 / desiredAspectRatio);
+        setHeightCalculated(heightCalculated);
+        setDesiredAspectRatio(desiredAspectRatio);
+
+      } else {
+        const ScreenWidth = Dimensions.get("window").width;
```

Comment:
```suggestion
      const screenWidth = Dimensions.get("window").width;
```
1. Rename variable
2. Why are we calculating screenWidth again and again, calculate once and store it for better performance. Change this approach.